pkg/image/external/external.go

```diff
@@ -0,0 +1,140 @@
+package external
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/rancher/rancher/pkg/controllers/management/k3sbasedupgrade"
+	"github.com/sirupsen/logrus"
+)
+
+type Source string
+
+const (
+	RKE2    Source = "rke2"
+	K3S     Source = "k3s"
+	UNKNOWN Source = "unknown"
+)
+
+func GetExternalUpgradeImages(rancherVersion string, externalData map[string]interface{}, source Source, os string, arch string) []string {
+	if source == "" {
+		source = UNKNOWN
+	}
```

Comment:
This function did not return an error in its original implementation, and this PR maintains that behavior. An `UNKNOWN` source is valid for this function, but will log a failure when download is performed.